### PR TITLE
8283624: Create an automated regression test for RFE-4390885

### PR DESCRIPTION
**Pull request by manukumarvs**
Write a regression test for [JDK-4390885](https://bugs.openjdk.java.net/browse/JDK-4390885) Enhancement Request.

Issue:
Please add the ability to set the location of a JFileChooser. This might be a
bug, but JFileChooser.setLocation() has no effect when
JFileChooser.showXXXXDialog() is called. This is becoming very important as the
popularity of multiple monitors is increasing. These dialogs show up on the
wrong monitor which is really annoying. Also due to bug #4189244 I am unable to
avoid painting problems by positioning the dialog away from the menu item that
initiated it.

Fix:
Now it's possible to set the location of Dialog in JFileChooser.

Testing:
1. Tested in Mach5 10 times per platform(macos,linux,windows) and got all Pass.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283624](https://bugs.openjdk.java.net/browse/JDK-8283624): Create an automated regression test for RFE-4390885

### Reviewers
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7996/head:pull/7996` \
`$ git checkout pull/7996`

Update a local copy of the PR: \
`$ git checkout pull/7996` \
`$ git pull https://git.openjdk.java.net/jdk pull/7996/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7996`

View PR using the GUI difftool: \
`$ git pr show -t 7996`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7996.diff">https://git.openjdk.java.net/jdk/pull/7996.diff</a>

</details>

### CODE REVIEW

**Comment by manukumarvs** [comment]
Any volunteers ?

**Comment by manukumarvs** [comment]
@aivanov-jdk Can you review this.

**Comment by manukumarvs** [comment]
Any volunteers?

File: test/jdk/javax/swing/JFileChooser/JFileChooserSetLocationTest.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+                if (!lafSetSuccess.get()) {
+                    continue;
+                }
+                robot.waitForIdle();
+
+                AtomicReference<Point> pt = new AtomicReference<>();
+                SwingUtilities.invokeAndWait(
+                        () -> pt.set(panel.getLocationOnScreen()));
+                Point panel_loc = pt.get();
+                xIn = (panel_loc.x + panel.getSize().width) / 2;
```
Comment (mrserb) [inline test/jdk/javax/swing/JFileChooser/JFileChooserSetLocationTest.java:103]: The getSize should be called on EDT as well.

Comment (manukumarvs) [reply test/jdk/javax/swing/JFileChooser/JFileChooserSetLocationTest.java]: Changed.

**Review by mrserb — APPROVED** [review]

**Pull request closed without merging**